### PR TITLE
Attributes Are Not Data Elements

### PR DESCRIPTION
**Pull request by mayerm94**
Make the attributes defined in the model-info (FacilityLoction and Component) extend from a base attribute instead of extending from DataElement
Pull requests into cqm-models require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable ("N/A") and :white_check_mark:.

**Submitter:**
- [x] This pull request describes why these changes were made.
- [x] Internal ticket for this PR:
- [x] Internal ticket links to this PR https://jira.mitre.org/browse/CYPRESS-418
- [x] Code diff has been done and been reviewed
- [x] Tests are included and test edge cases
- [x] Tests have been run locally and pass
- [x] Cqm-execution fixtures have been updated with the update_cqm_execution_fixtures.sh script inside server-scripts using this branch in the cqm-converter N/A no diff in fixtures

**Bonnie Reviewer:**

Name:
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

**Cypress Reviewer:**

Name:
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

### CODE REVIEW

File: app/models/qdm/attributes/attribute.rb [hunk 1]
```diff
@@ -0,0 +1,64 @@
+module QDM
+  # Represents QDM attribute
+  class Attribute
+    include Mongoid::Document
+    embedded_in :data_element
```
Comment (losborne) [inline app/models/qdm/attributes/attribute.rb]: Why is a QDM::Attribute `embedded_in` a `:data_element`?  `DataElement` does not have an `embeds_one` or `embeds_many` on it.

File: app/models/qdm/attributes/attribute.rb [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+      json_representation
+    end
+
+    # Include '_type' in any JSON output. This is necessary for deserialization.
+    def to_json(options = nil)
+      serializable_hash(methods: :_type).to_json(options)
+    end
+
+    class << self
+    # Get the object as it was stored in the database, and instantiate
```
Comment (losborne) [inline app/models/qdm/attributes/attribute.rb]: There should be another indentation level starting here

File: app/models/qdm/basetypes/data_element.rb [hunk 3]
```diff
@@ -66,10 +66,18 @@ def shift_dates(seconds)
         end
 
         # Special case for facility locations
-        next unless field == 'facilityLocations'
-        send(field).each do |facility_location|
-          facility_location['locationPeriod'][:low] = (facility_location['locationPeriod'][:low].to_time + seconds).to_datetime
-          facility_location['locationPeriod'][:high] = (facility_location['locationPeriod'][:high].to_time + seconds).to_datetime
+        if field == 'facilityLocations'
+          send(field).each do |facility_location|
+            facility_location['locationPeriod'][:low] = (facility_location['locationPeriod'][:low].to_time + seconds).to_datetime
```
Comment (losborne) [inline app/models/qdm/basetypes/data_element.rb]: These 2 lines are duplicated below and should be pulled out into a separate function.

File: app/models/qdm/attributes/attribute.rb (hunk 1 above)
Comment (mayerm94) [reply app/models/qdm/attributes/attribute.rb]: Forgotten Artifact from experimentation -- good catch

**Pull request merged**